Add third column to specific emplacements table describing relationship

rcousens opened the pull request:
The standard Symfony2 framework has mandatory requirements about specific emplacements, and then others that are considered best practice by convention. This commit makes the documentation clearer in order to provide a developer with reasonable expectations about the consequences of altering the default directory structure. Additionally, newer developers may be less intimidated and concerned by whether they are compromising any auto-wiring functionality by deviating from the information provided.

Fixes #5502

rcousens committed:
Add third column to specific emplacements table describing relationship

The standard Symfony2 framework has mandatory requirements about specific emplacements, and then others that are considered best practice by convention. This commit makes the documentation clearer in order to provide a developer with reasonable expectations about the consequences of altering the default directory structure. Additionally, newer developers may be less intimidated and concerned by whether they are compromising any auto-wiring functionality by deviating from the information provided.

javiereguiluz commented:
:+1: thank you @rcousens.

The only thing that I don't like is the column title (`Relationship`). I propose to change it by `Location Type` or something like that. But before doing any change, let's see what do doc managers think about this.

rcousens commented:
Sure thing, I also think it could use a little extra padding as it looks a bit cramped! Will wait for feedback and make the final changes then.

WouterJ commented on cookbook/bundles/best_practices.rst:
```diff
@@ -117,20 +117,20 @@ files are going to be part of the repository.
+Web Resources (CSS, JS, images)  ``Resources/public/``         Mandatory
+Translation files                ``Resources/translations/``   Mandatory
+Templates                        ``Resources/views/``          Mandatory
+Unit and Functional Tests        ``Tests/``                    Convention
```
in some degree, this is mandatory too (as it's configured in the phpunit config file).

rcousens committed:
Fix as per comments

Renamed Relationship column to Location and updated value for Tests to Mandatory due to default phpunit config.

rcousens commented:
Updated as per feedback

javiereguiluz commented:
:+1: @rcousens thanks for your patience and for making all the asked changes! This is ready to be merged :)

PS: @WouterJ when merging this, please add two whitespaces as separation between the second and the new third column. Thanks!

rcousens committed:
Fix spacing on third column

rcousens commented:
@WouterJ I fixed up the spacing :) All good! 

Thank you for letting me improve the documentation, I remember starting with Symfony2 almost three years ago now and the amount of internal anxiety I had over "specific emplacements" made me hesitant to structure bundles in a way that best expressed the domain. Knowing what is mandatory and what is optional is somewhat liberating when you finally work it out.

Cheers :+1:

xabbuh commented:
Hm, thinking about this using only "Location" sounds weird to me. I would prefer "Location Type" or something like that instead.

stof commented on cookbook/bundles/best_practices.rst:
```diff
@@ -117,20 +117,20 @@ files are going to be part of the repository.
+Web Resources (CSS, JS, images)  ``Resources/public/``          Mandatory
+Translation files                ``Resources/translations/``    Mandatory
+Templates                        ``Resources/views/``           Mandatory
+Unit and Functional Tests        ``Tests/``                     Mandatory
```
this is not really mandatory. It is perfectly fine to put them elsewhere (even outside the prod code of your bundle) as long as you configure your testing tool accordingly (and it is perfectly fine to use another tool than PHPUnit, in which case tests would be elsewhere)

rcousens commented:
I will get this fixed up again this weekend. Cheers. Any preference on the third column header name?

Location vs Location Type?

WouterJ commented:
Personally, I find the values of the last column too long. I propose to change it to use `Yes`/`No` as value:

![sf-best-practices-mandatory](https://cloud.githubusercontent.com/assets/749025/9703940/84b84e1e-5493-11e5-94fe-c9ea6cedf0cc.png)

Or leave the column empty, unless the option is mandatory:

![sf-best-practices-mandatory](https://cloud.githubusercontent.com/assets/749025/9703960/85579284-5494-11e5-8946-320e3ffed852.png)

javiereguiluz commented:
@rcousens I've created #6616 to fix your work applying the comments made by @Wouter. I've reused your original commits, so you'll get full credit for your work. Thanks!

xabbuh commented:
I am closing here in favour of #6616. Thank you for starting this change @rcousens!
